test/hotspot/jtreg/runtime/cds/appcds/complexURI/ComplexURITest.java

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024 JetBrains s.r.o.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Verifies that CDS works with jar located in directories
+ *          with names that need escaping
+ * @bug 8339460
+ * @requires vm.cds
+ * @requires vm.cds.custom.loaders
+ * @requires vm.flagless
+ * @library /test/lib /test/hotspot/jtreg/runtime/cds/appcds
+ * @compile mypackage/Main.java mypackage/Another.java
+ * @run main/othervm ComplexURITest
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class ComplexURITest {
+    final static String listFileName = "test-classlist.txt";
+    final static String archiveName  = "test-dynamic.jsa";
```

Comment:
You're using the same `archiveName` for both static and dynamic test cases. I'd suggest declare a different archive name for the static case. E.g.
```
final static String staticArchiveName = "test-static.jsa";
final static String dynamicArchiveName = "test-dynamic.jsa";
```
Also, I think this test should be excluded from the `hotspot_appcds_dynamic` test group in the `open/test/hotspot/jtreg/TEST.groups` file.

Reply:
Done. I also added the code to safeguard against `IOException` on Windows in the case of deleting the read-only archive created previously.

Reply:
Test changes look good. I will run some testing on your patch.